vertical-pod-autoscaler/pkg/utils/limitrange/limit_range_calculator.go

```diff
@@ -64,19 +65,47 @@ func NewNoopLimitsCalculator() *noopLimitsRangeCalculator {
 	return &noopLimitsRangeCalculator{}
 }
 
-func (lc *limitsChecker) GetContainerLimitRangeItem(namespace string) (*v1.LimitRangeItem, error) {
+func (lc *limitsChecker) GetContainerLimitRangeItem(namespace string) (*core.LimitRangeItem, error) {
 	limitRanges, err := lc.limitRangeLister.LimitRanges(namespace).List(labels.Everything())
 	if err != nil {
 		return nil, fmt.Errorf("error loading limit ranges: %s", err)
 	}
 
+	updatedResult := func(result core.ResourceList, lrItem core.ResourceList,
+		resourceName core.ResourceName, comparator func(q1, q2 resource.Quantity) bool) core.ResourceList {
+		if lrItem == nil {
+			return result
+		}
+		if result == nil {
+			return lrItem.DeepCopy()
+		}
+		if lrResource, lrHas := lrItem[resourceName]; lrHas {
+			resultResource, resultHas := result[resourceName]
+			if !resultHas || comparator(resultResource, lrResource) {
+				result[resourceName] = lrResource
+			}
+		}
+		return result
+	}
+	maxComparator := func(q1, q2 resource.Quantity) bool { return q1.Cmp(q2) > 0 }
+	minComparator := func(q1, q2 resource.Quantity) bool { return q1.Cmp(q2) < 0 }
```

Comment:
My initial impression was that those do exact opposite of what they do. Maybe rename compactor returning bool to value picker that returns new value (if both are present)? It will be a little longer but easier to understand.

Reply:
Yeah, I thought this might be confusing. Changed.